### PR TITLE
Add autoRtl prop to Popover

### PR DESCRIPTION
**Pull request by sirreal**
This is a proposal implementation to add left-to-right or right-to-left (LTR/RTL) based logic to Popover positioning.

Because Popovers are positioned in a way that is relative to their context and the content surrounding them, it is reasonable to assume that LTR/RTL directionality should invert their left/right positioning.

This PR proposes such an implementation, automatically adjusting a provided position (assumed to be LTR), and adjusting it as required based on the `isRtl` selector.

See p4TIVU-6tD-p2

There are two points from the post that would be good to discuss and finalize:

* Naming: is `autoRtl` a good name for the prop?
  It *does* automatically handle the RTL case by swapping right and left values. The prop could try to reference this with something like direction, reflect, etc.
* Default. Should this be `true` by default?

This was initially mentioned in #12278

### CODE REVIEW

**Comment by sirreal** [comment]
I'm curious about some logic related to positioning suggestions. `bottom left` seems to be treated differently from `left bottom` while README/devdocs don't even include left/right-first positions. Seems like they would be discouraged aliases.

@retrofox can you provide any context or background on the left/right-first positioning?

**Comment by sirreal** [comment]
Flagging for i18n review to get input on whether this behaviour appears correct in RTL contexts.

File: client/components/popover/index.jsx [hunk 1]
```diff
@@ -260,6 +274,27 @@ class Popover extends Component {
...
+	 * Adjusts positition swapping left and right values
+	 * when right-to-left directionality is found.
+	 *
+	 * @param  {String} position Original position
+	 * @return {String}          Adjusted position
+	 */
+	adjustRtlPosition( position ) {
+		if ( this.props.isRtl ) {
+			switch ( position ) {
+				case 'top right': return 'top left';
```
Comment (roccotripaldi) [inline client/components/popover/index.jsx]: This is minor nit-pick, but we may want returns on a new line.

**Comment by yoavf** [comment]
> Flagging for i18n review to get input on whether this behaviour appears correct in RTL contexts.
It does - thanks for working on this!

File: client/components/info-popover/index.jsx [hunk 2]
```diff
@@ -11,84 +11,85 @@ import Popover from 'components/popover';
 import classNames from 'classnames';
 import analytics from 'lib/analytics';
 
-export default React.createClass( {
+export default class InfoPopover extends Component {
+	static propTypes = {
+		autoRtl: PropTypes.bool,
```
Comment (tyxla) [inline client/components/info-popover/index.jsx]: Thanks for alphabetically sorting the props ❤️ 

Maybe also alphabetically sort the list of suppoted positions? :trollface:

File: client/components/info-popover/index.jsx [hunk 3]
```diff
@@ -11,84 +11,85 @@ import Popover from 'components/popover';
...
-		gaEventCategory: React.PropTypes.string,
-		popoverName: React.PropTypes.string,
-		ignoreContext: React.PropTypes.shape( {
-			getDOMNode: React.PropTypes.function
-		} ),
-	},
+	handleClick = ( event ) => {
+		event.preventDefault();
+		this.setState( {
+			showPopover: ! this.state.showPopover },
```
Comment (tyxla) [inline client/components/info-popover/index.jsx]: Indentation/formatting seems broken here, can we fix it? I'd expect something like:

```
this.setState( {
	showPopover: ! this.state.showPopover
}, this.recordStats );
```

File: client/components/info-popover/index.jsx (hunk 2 above)
Comment (sirreal) [inline client/components/info-popover/index.jsx]: 😂  I did CSS-style clockwise from `top` ordering!

File: client/components/info-popover/index.jsx (hunk 3 above)
Comment (sirreal) [inline client/components/info-popover/index.jsx]: Yuck, thanks for catching that!

File: client/components/popover/index.jsx (hunk 1 above)
Comment (sirreal) [inline client/components/popover/index.jsx]: Added some _extra_ spacing _and_ decided to be nice and add the non-canonical `left`/`right`-first positions. Thanks!

File: client/components/info-popover/index.jsx [hunk 4]
```diff
@@ -11,84 +11,82 @@ import Popover from 'components/popover';
...
+	static propTypes = {
+		autoRtl: PropTypes.bool,
+		className: PropTypes.string,
+		gaEventCategory: PropTypes.string,
+		id: PropTypes.string,
+		ignoreContext: PropTypes.shape( {
+			getDOMNode: PropTypes.func
+		} ),
+		popoverName: PropTypes.string,
+		position: PropTypes.oneOf( [
```
Comment (tyxla) [inline client/components/info-popover/index.jsx]: @sirreal I'm noticing that there are usages of `position` that don't match these `propTypes`:



To reproduce, go to http://calypso.localhost:3000/plans/$site where $site is a WordPress.com site.

Comment (sirreal) [inline client/components/info-popover/index.jsx]: That's exactly the kind of thing I was trying to catch with the stricter propType. The positioning behaviour of that popover differs from other popovers in [some](https://github.com/Automattic/wp-calypso/pull/13600#issuecomment-298934782) [unspecified](https://github.com/Automattic/wp-calypso/blob/9e631355f68b8d6ac864b6bf3f12d6bb61c071fe/client/components/popover/util.js#L279-L284) [way](https://github.com/Automattic/wp-calypso/blob/9e631355f68b8d6ac864b6bf3f12d6bb61c071fe/client/components/popover/util.js#L258-L263) and doesn't adhere to [the docs](https://github.com/Automattic/wp-calypso/blob/9e631355f68b8d6ac864b6bf3f12d6bb61c071fe/client/components/popover/README.md#position--string----default-top).

I've created #14403 to correct this case. After a quick grep of the codebase, it seems you came across the _only_ place this occurrs 🙂

Comment (tyxla) [inline client/components/info-popover/index.jsx]: Thank you sir!

**Pull request merged**